lib/puppet/provider/cisco_interface/cisco.rb

```diff
@@ -203,24 +184,36 @@ def self.properties_get(interface_name, nu_obj)
     new(current_state)
   end # self.properties_get
 
-  def self.instances
+  def self.instances(single_intf=nil)
+    # 'puppet resource' calls here directly; will always get all interfaces.
+    # 'puppet agent' callpath is initialize->prefetch; may pass a single intf.
+    all_intf = single_intf ? false : true
     interfaces = []
-    Cisco::Interface.interfaces.each do |interface_name, nu_obj|
+    Cisco::Interface.interfaces(nil, single_intf).each do |interface_name, nu_obj|
       begin
         # Some interfaces cannot or should not be managed by this type.
         # - NVE Interfaces (managed by cisco_vxlan_vtep type)
         next if interface_name.match(/nve/i)
-        interfaces << properties_get(interface_name, nu_obj)
+        interfaces << properties_get(interface_name, nu_obj, all_intf: all_intf)
       end
     end
     interfaces
   end # self.instances
 
   def self.prefetch(resources)
-    interfaces = instances
-    resources.keys.each do |name|
-      provider = interfaces.find { |intf| intf.instance_name == name }
-      resources[name].provider = provider unless provider.nil?
+    if resources.keys.length > 10
+      interfaces = instances
+      info "[prefetch all-interfaces: #{interfaces.length}]"
+      resources.keys.each do |name|
+        provider = interfaces.find { |intf| intf.instance_name == name }
+        resources[name].provider = provider unless provider.nil?
+      end
+    else
+      info '[prefetch per-interface]'
```

Comment:
This `info` statement confuses me.  I assume this means prefetch one interface at a time?

Reply:
fixed